### PR TITLE
Use conda-build-all for building recipes

### PR DESCRIPTION
**Pull request by dkillick**
Replace `obvious-ci` with `conda-build-all` in `.travis.yml` ~~and `circle.yml`~~.

### CODE REVIEW

**Comment by dkillick** [comment]
I've reverted the change to circle CI as it turned out to be a larger change than it initially appeared. The change will reappear in a followup PR. I've re-targeted this to be for **Travis CI only**.

**Comment by dkillick** [comment]
Winning! The optional upload flag has got the checks passing. Ping @pelson

File: .travis.yml [hunk 1]
```diff
@@ -30,14 +30,17 @@ before_install:
 install:
     # Set the numpy variable. This isn't used, but conda-build complains if we haven't set it already.
     - export CONDA_NPY=19
+    # Set the name of the channel label to upload to.
+    - export LABEL_NAME=main
+    # Only run the upload if the binstar token is present.
+    - export UPLOAD_CHANNELS=$(if [ -z ${BINSTAR_TOKEN+x} ]; then echo ""; else echo "--upload-channels scitools/label/${LABEL_NAME}"; fi)
```
Comment (pelson) [inline .travis.yml]: Options:

```UPLOAD_CHANNEL=$([ -z ${WIBBLE+x} ] || echo "foobar")```

```
UPLOAD_CHANNEL=""
[ -z ${WIBBLE+x} ] || UPLOAD_CHANNEL="foobar"
```

**Comment by pelson** [comment]
Great. Thanks for this @dkillick.

**Pull request merged**